[ARM] Generate build-attributes more correctly in the presence of intrinsic declarations.

davemgreen opened the pull request:
This code doesn't work very well, but this makes it work when intrinsic definitions are present. It now discounts functions declarations from the set of attributes it looks at.

The code would have worked better before 0ab5b5b8581d9f2951575f7245824e6e4fc57dec when module-level attributes could provide the information used to construct build-attributes.

smithp35 commented on llvm/lib/Target/ARM/ARMAsmPrinter.cpp:
```diff
@@ -610,20 +610,23 @@ void ARMAsmPrinter::emitEndOfAsmFile(Module &M) {
 
- // Returns true if all functions have the same function attribute value.
- // It also returns true when the module has no functions.
+// Returns true if all functions have the same function attribute value.
```
Suggest a small comment update, also could be applied on line 623
```
Returns true if all function definitions have the same function attribute value.
```

smithp35 reviewed (approved):
LGTM. I agree that we should skip over declarations when considering the build attributes for this module. The module defining the declared functions will define its build attributes.

davemgreen committed:
[ARM] Generation build-attributes more correctly in the presence of function declarations.

This code doesn't work very well, but this at least makes it work when
intrinsic definitions are present. It now discounts functions declarations from
the set of attributes it looks at. It would have worked better before
0ab5b5b8581d9f2951575f7245824e6e4fc57dec when module-level attributes could
provide the information used to construct build-attributes.
